internal/broker/broker.go

```diff
@@ -669,38 +749,22 @@ type claims struct {
 	Sub               string `json:"sub"`
 }
 
-func (b *Broker) userInfoFromClaims(userClaims claims, groups []group.Info) (string, error) {
-	user := struct {
-		Name   string
-		UUID   string
-		Home   string
-		Shell  string
-		Gecos  string
-		Groups []group.Info
-	}{
+type userInfo struct {
+	Name   string       `json:"name"`
+	UUID   string       `json:"uuid"`
+	Home   string       `json:"dir"`
+	Shell  string       `json:"shell"`
+	Gecos  string       `json:"gecos"`
+	Groups []group.Info `json:"groups"`
+}
+
+func (b *Broker) userInfoFromClaims(userClaims claims, groups []group.Info) userInfo {
+	return userInfo{
 		Name:   userClaims.Email,
 		UUID:   userClaims.Sub,
 		Home:   filepath.Join(b.homeDirPath, userClaims.Email),
 		Shell:  "/usr/bin/bash",
 		Gecos:  userClaims.Name,
 		Groups: groups,
 	}
-
-	var buf bytes.Buffer
-	err := template.Must(template.New("").Parse(`{
-		"name": "{{.Name}}",
-		"uuid": "{{.UUID}}",
-		"gecos": "{{.Gecos}}",
-		"dir": "{{.Home}}",
-		"shell": "{{.Shell}}",
-		"groups": [ {{range $index, $g := .Groups}}
-			{{- if $index}}, {{end -}}
-			{"name": "{{.Name}}", "ugid": "{{.UGID}}"}
-		{{- end}} ]
-}`)).Execute(&buf, user)
-	if err != nil {
-		return "", err
-	}
```

Comment:
IIRC, the templating was necessary due to the way `json.Marshal` formats go slices, but I don't recall exactly the reason why. Can you give some more input here, @didrocks?

Reply:
The issue was with 0 length or nil slices IIRC, but to be confirmed.